gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/medication/MedicationRequestMapper.java

```diff
@@ -36,24 +49,151 @@ public class MedicationRequestMapper extends AbstractMapper<DomainResource> {
         String practiseCode) {
         try {
-            return mapEhrExtractToFhirResource(ehrExtract, (extract, composition, component) ->
-                    extractAllMedications(component)
-                            .filter(Objects::nonNull)
-                            .flatMap(medicationStatement -> mapMedicationStatement(
-                                    ehrExtract, composition, medicationStatement, patient, encounters, practiseCode)))
-                    .toList();
+            var resources = mapEhrExtractToFhirResource(ehrExtract, (extract, composition, component) ->
+                extractAllMedications(component)
+                    .filter(Objects::nonNull)
+                    .flatMap(medicationStatement -> mapMedicationStatement(
+                        ehrExtract, composition, medicationStatement, patient, encounters, practiseCode)))
+                .collect(Collectors.toCollection(ArrayList::new));
+
+            var acutePlans = getAcutePlans(resources);
+            var ordersGroupedByAcutePlan = getOrdersGroupedByReferencedAcutePlan(resources, acutePlans);
+
+            ordersGroupedByAcutePlan
+                .forEach((plan, orders) -> {
+                    if (orders.size() > 1) {
+                        sortOrdersByValidityPeriodStart(orders);
+
+                        // the index starts at one as the earliest order remains referenced to the original plan
+                        for (var index = 1; index < orders.size(); index++) {
+                            var duplicatedPlan = plan.copy();
+
+                            // we need to update the id and identifier to the generated id;
+                            var duplicatedPlanId = idGeneratorService.generateUuid();
+                            duplicatedPlan.setId(duplicatedPlanId);
+                            duplicatedPlan.getIdentifierFirstRep().setValue(duplicatedPlanId);
+
+                            // we need to update the basedOn of the order to match the generated plan
+                            var basedOn = orders.get(index).getBasedOn();
+                            updateBasedOnReferenceToReferenceDuplicatedPlan(basedOn, duplicatedPlanId);
+
+                            // we need to set the prior medication ref to either the initial plan (when index == 1)
+                            // or to id of the previously generated plan (that is the plan referenced index -1)
+                            var previousOrderBasedOn = orders.get(index - 1).getBasedOn();
+                            var previousBasedOnReference = getMedicationRequestBasedOnReference(previousOrderBasedOn);
+                            previousBasedOnReference.ifPresent(duplicatedPlan::setPriorPrescription);
+
+                            resources.add(duplicatedPlan);
+                        }
+                    }
+                });
+
+            return resources;
         } finally {
             medicationMapperContext.reset();
         }
     }
```

Comment:
This is quite a large method, in the spirit of keeping functions small - an [interesting article](https://martinfowler.com/bliki/FunctionLength.html).